pkg/build/api/v1/conversion.go

```diff
@@ -166,17 +166,27 @@ func convert_api_BuildSource_To_v1_BuildSource(in *newer.BuildSource, out *Build
 	if err := s.DefaultConvert(in, out, conversion.IgnoreMissingFields); err != nil {
 		return err
 	}
+	typeFound := false
 	switch {
+	case len(in.Images) > 0:
+		out.Type = BuildSourceImage
+		typeFound = true
```

Comment:
Ah i had you said the opposite earlier but yeah putting it at the end makes
sense

On Thursday, April 7, 2016, Ben Parees <notifications@github.com> wrote:

> In pkg/build/api/v1/conversion.go
> <https://github.com/openshift/origin/pull/8219#discussion_r58973866>:
>
> >  	switch {
> > +	case len(in.Images) > 0:
> > +		out.Type = BuildSourceImage
> > +		typeFound = true
>
> the way it is now, any build w/ images>0 will be labeled as
> BuildSourceImage.
>
> you want this to be the last check in the case. (basically the cases be
> listed in priority order: Git, Binary, Dockerfile, SourceImage)
>
> you can also make the out.Type=BuildSourceNone be the default case for the
> switch, rather than a separate if.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8219/files/b4e4fb1144d9e8493e9d4564a777ea263fa3eaac#r58973866>
>